repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala

```diff
@@ -62,9 +62,47 @@ object ProjectUpdateRequest {
   implicit val updateRequestFormat: OFormat[ProjectUpdateRequest] = Json.format[ProjectUpdateRequest]
 }
 
-final case class Version(name: String, message: String, target: String, commit: Commit)
-object Version {
-  implicit val versionPlayFormat: OFormat[Version] = Json.format[Version]
+final case class GitLabVersion(name: PipelineVersion, message: String, target: String, commit: Commit)
+object GitLabVersion {
+  implicit val gitlabVersionFormat: OFormat[GitLabVersion] = Json.format[GitLabVersion]
+}
+
+final case class PipelineVersion(major: VersionValue, minor: VersionValue, revision: VersionValue)
+    extends Ordered[PipelineVersion] {
+  import VersionValue._
+  def name: String = s"v$major.$minor.$revision"
+
+  private val ordering: Ordering[PipelineVersion] =
+    Ordering[(VersionValue, VersionValue, VersionValue)].on((pv: PipelineVersion) => (pv.major, pv.minor, pv.revision))
+
+  override def compare(that: PipelineVersion): Int =
+    ordering.compare(this, that)
+
+  def increaseMajor: PipelineVersion =
+    this.copy(major = increment(this.major))
+
+  def increaseMinor: PipelineVersion =
+    this.copy(minor = increment(this.minor))
+
+  def increaseRevision: PipelineVersion =
+    this.copy(revision = increment(this.revision))
+
+  override def toString: String = this.name
+}
+
+object PipelineVersion {
+  import VersionValue._
+  private val pattern = """^v(\d+)\.(\d+)\.(\d+)$""".r
```

Comment:
This pattern assumes the inner structure of the `VersionValue`. However, this structure must be hidden even for `PipelineVersion`
You must check that the string starts with `v` and contains three subgroups separated by `.`. Each of these subgroups must be validated inside `VersionValue.fromString` method, rather than here.